### ISSUE TITLE
qwen3 vl finetune

### ISSUE THREAD

**Issue by baochi0212** [modelscope/ms-swift#6207]
I train qwen 3 vl moe version (30b, 220b) and seems to converge well on my data. with below log and script but the lora checkpoints return the same output with base, I try the 2.5 version not bug :) 
- Final steps log: 
{"loss": 0.07646036148071289, "grad_norm": 2.7671284675598145, "learning_rate": 6.293616306246586e-09, "token_acc": 0.9905684263830865, "epoch": 9.95, "step": 199}, {"loss": 0.08132177591323853, "grad_norm": 6.692193984985352, "learning_rate": 0.0, "token_acc": 0.9904822335025381, "epoch": 10.0, "step": 200}, {"train_runtime": 1605.5379, "train_samples_per_second": 3.843, "train_steps_per_second": 0.125, "total_flos": 4.910286200058675e+17, "train_loss": 2.025656066536903, "epoch": 10.0, "step": 200}], "memory": 74.98828125}
- script:
MODEL_TYPE="qwen3_vl"
DEEPSPEED=zero3
pip install -U qwen_vl_utils
if [[ $MODEL_NAME == *_72b ]]; then
    BS=2
else
    BS=4
fi
echo "MODEL TYPE: $MODEL_TYPE, Batch size: $MODEL_NAME, $BS"
VISION_LR=$2
LLM_LR=$3
N_EPOCHS=$4
PACKING=$5
# main
IMAGE_MAX_TOKEN_NUM=1024 \
NPROC_PER_NODE=8 \
swift sft \
    --model $MODEL_NAME \
    --train_type lora \
    --use_hf 1 \
    --model_type $MODEL_TYPE \
    --dataset "/prj/corp/airesearch/lasvegas/vol5-scratch/users/chitb/data/hf_screen/${6}_data" \
    --load_from_cache_file false \
    --split_dataset_ratio 0.0 \
    --torch_dtype bfloat16 \
    --attn_impl "flash_attn" \
    --packing $PACKING \
    --padding_free $PACKING \
    --num_train_epochs $N_EPOCHS \
    --per_device_train_batch_size 1 \
    --gradient_accumulation 1 \
    --per_device_eval_batch_size 1 \
    --vit_lr $VISION_LR \
    --aligner_lr $LLM_LR \
    --learning_rate $LLM_LR \
    --lora_rank 128 \
    --lora_alpha 256 \
    --target_modules all-linear \
    --router_aux_loss_coef 1e-3 \
    --gradient_accumulation_steps 1 \
    --gradient_checkpointing True \
    --eval_strategy "none" \
    --save_strategy "epoch" \
    --save_total_limit 1 \
    --logging_steps 1 \
    --report_to "none" \
    --max_length 4096 \
    --output_dir "output/${MODEL_TYPE}_${VISION_LR}_${LLM_LR}_${N_EPOCHS}epochs_${PACKING}_packing_${6}" \
    --warmup_ratio 0.01 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 8 \
    # --use_liger_kernel true \
    # --deepspeed $DEEPSPEED

**HARDWARE**
8 x H100 80GBs

**Context**
I try with huggingface trl and qwen codebase but qwen codebase full finetune seems not work but trl work for 30b, but larger memory required, 220b I can't fit atm 
Thanks in advance for discussion

**Comment by cxmscb** [comment]
same issue, maybe something error in the merge process.

**Comment by Jintao-Huang** [comment]
https://swift.readthedocs.io/en/latest/BestPractices/Qwen3-VL-Best-Practice.html

**Comment by baochi0212** [comment]
tks @Jintao-Huang  for quick rep, I try the swift infer with adapater checkpoint:
```
CUDA_VISIBLE_DEVICES=0 \
IMAGE_MAX_TOKEN_NUM=1024 \
VIDEO_MAX_TOKEN_NUM=128 \
FPS_MAX_FRAMES=16 \
swift infer \
    --adapter ... 
    --model ... \
    --stream true
```
it works well but when I use transformers generate, the lora not works (like in 2.5), what maybe the cause ?, tks
```
from transformers import AutoModelForImageTextToText, AutoProcessor, TextStreamer, Qwen3VLMoeForConditionalGeneration
import sys
import torch
from peft import PeftModel
model_name = sys.argv[1]
base_model = Qwen3VLMoeForConditionalGeneration.from_pretrained(
    sys.argv[2],
    dtype=torch.bfloat16,
    attn_implementation="flash_attention_2",
    device_map="auto",
)
model = PeftModel.from_pretrained(base_model, model_name)
# print(model)
model.merge_and_unload()
processor = AutoProcessor.from_pretrained(sys.argv[2])
streamer = TextStreamer(processor)
while True:
    image = input("Image path: ")
    text = input("Query: ")
    messages = [
        {
            "role": "user",
            "content": [
                {
                    "type": "image",
                    "image": image,
                },
                {"type": "text", "text": text},
            ],
        }
    ]

    # Preparation for inference
    inputs = processor.apply_chat_template(
        messages,
        tokenize=True,
        add_generation_prompt=True,
        return_dict=True,
        return_tensors="pt"
    )

    # Inference: Generation of the output
    generated_ids = model.generate(**inputs, 
                                   do_sample=False,
                                   streamer=streamer,
                                   max_new_tokens=128)
```

**Comment by Jintao-Huang** [comment]
set max_pixels in messages


tokens * 32 * 32

e.g. 1024 * 32 * 32

**Comment by baochi0212** [comment]
tks @Jintao-Huang , but I try to set both in processor and in messages, and try to process with apply_chat_template and process_vision but still generate the original result :) 
```
processor = AutoProcessor.from_pretrained(sys.argv[2], max_pixels=1024*32*32)
streamer = TextStreamer(processor)
while True:
    image = input("Image path: ")
    text = input("Query: ")
    messages = [
        {
            "role": "user",
            "content": [
                {
                    "type": "image",
                    "max_pixels": 1024*32*32,
                    "image": image,
                },
                {"type": "text", "text": text},
            ],
        }
    ]
```

**Comment by siri-rouser** [comment]
Hi baochi, 

I have met the same problem. Have you solve it out?

Best,
Yuqiang

**Comment by slin000111** [comment]
> Hi baochi,
> 
> I have met the same problem. Have you solve it out?
> 
> Best, Yuqiang

Please check if the input_ids of Swift infer command-line inference and PEFT inference are consistent.
